File: python/7 kyu/376.py
# https://www.codewars.com/kata/5d6f49d85e45290016bf4718/train/python

# Passed

def any_odd(x):
    binary_num = bin(x).split('b')[-1]
    result = any(bit == "1" and idx % 2 == 1 for idx, bit in enumerate(binary_num[::-1]))
    return result

output = any_odd(131)
print(output)

File: nodejs/7 kyu/376.js
// https://www.codewars.com/kata/6a520b948dc07d11d0342a58/train/javascript

// Passed

function zfcDefinition(n, memo = {}) {
  if (n === 0) {
    return "{}";
  }

  let definitions = [];
  for (let num = 0; num < n; num++) {
    let definition;
    if (num in memo) {
      definition = memo[num];
    } else {
      definition = zfcDefinition(num, (memo = memo));
      memo[num] = definition;
    }

    definitions.push(definition);
  }
  let result = `{${definitions.join(",")}}`;
  return result;
}

const output = zfcDefinition(3);
console.log(output);